app/forms/state_file/esign_declaration_form.rb

```diff
@@ -38,6 +38,7 @@ def save
       old_efile_submission = @intake.efile_submissions&.last
       if old_efile_submission.present?
         # the after_transitions :resubmission creates a new efile submission and transitions it to :preparing
+        @intake.submission_pdf.detach
```

Comment:
i think this deserves a comment on why we're detaching here. the way you explained in the PR description makes perfect sense, but i don't think i'll remember that six months from now.